src/helpers/user-assigned-timespans.ts

```diff
@@ -0,0 +1,67 @@
+import { IssueParams } from "../start";
+import { ContextPlugin } from "../types/plugin-input";
+import { IssueActivity } from "../issue-activity";
+
+export interface AssignmentPeriod {
+  assignedAt: string;
+  unassignedAt: string | null;
+}
+
+export interface UserAssignments {
+  [username: string]: AssignmentPeriod[];
+}
+
+/*
+ * Returns the list of assignment periods per user for a given issue.
+ */
+export async function getAssignmentPeriods(octokit: ContextPlugin["octokit"], issueParams: IssueParams) {
+  const events = await octokit.paginate(octokit.rest.issues.listEvents, {
+    ...issueParams,
+    per_page: 100,
+  });
+
+  const userAssignments: UserAssignments = {};
+
+  events
+    .filter((event) => ["assigned", "unassigned"].includes(event.event))
+    .forEach((event) => {
+      const username = "assignee" in event ? event.assignee?.login : null;
+      if (!username) return;
+
+      if (!userAssignments[username]) {
+        userAssignments[username] = [];
+      }
+
+      const lastPeriod = userAssignments[username][userAssignments[username].length - 1];
+
+      if (event.event === "assigned") {
+        userAssignments[username].push({
+          assignedAt: event.created_at,
+          unassignedAt: null,
+        });
+      } else if (event.event === "unassigned" && lastPeriod && lastPeriod.unassignedAt === null) {
+        lastPeriod.unassignedAt = event.created_at;
```

Comment:
This logic isn't clear to me. 
- I see if it doesn't find an unassign event at the end that it will use the last assigned event as the timestamp (which doesn't make sense to me) 
- How do we derive the unassign timestamp? 
- What if they make it to completion of the task? 
- What if their assignment is toggled multiple times? Are we excluding all comments between?

Reply:
It is set that way so all the comments are taken out between assignment and completion. Currently it only takes out the comment during assignment, meaning if they get un-assigned they would be counted inbetween.